GameServer.cpp

```diff
@@ -144,18 +143,16 @@ int GameServer::AllocSocket(SOCKET& hostSocket, unsigned short port)
 	int errorCode = bind(hostSocket, (sockaddr*)&hostAddress, sizeof(hostAddress));
 
 	if (errorCode == SOCKET_ERROR) {
-		return SocketBindFailed;
+		throw std::runtime_error("Unable to bind socket");

```

Comment:
I would recommend adding a formatted address to the error message (which includes the port).